src/steadystate_iterative.jl

```diff
@@ -0,0 +1,147 @@
+using ..timeevolution: nh_hamiltonian, dmaster_h, dmaster_nh, check_master
+
+"""
+    iterative!(rho0, H, J, [method!], args...; [log=false], kwargs...) -> rho[, log]
+
+Compute the steady state density matrix of master equation defined by a
+Hamiltonian and a set of jump operators by solving `L rho = 0` via an iterative
+method provided as argument.
+
+# Arguments
+* `rho0`: Initial density matrix. Note that this gets mutated in-place.
+* `H`: Operator specifying the Hamiltonian.
+* `J`: Vector containing all jump operators.
+* `method!`: The iterative method to be used. Defaults to `IterativeSolvers.bicgstabl!`
+    or `IterativeSolvers.idrs!` depending on arguments' types.
+* `rates=nothing`: Vector or matrix specifying the coefficients (decay rates) for
+    the jump operators. If nothing is specified all rates are assumed to be 1.
+* `Jdagger=dagger.(J)`: Vector containing the hermitian conjugates of the jump
+    operators. If they are not given they are calculated automatically.
+* `args...`: Further arguments are passed on to the iterative solver.
+* `kwargs...`: Further keyword arguments are passed on to the iterative solver.
+See also: [`iterative`](@ref)
+
+Credit for this implementation goes to Z. Denis and F. Vicentini.
+See also https://github.com/Z-Denis/SteadyState.jl
+"""
+function iterative!(rho0::Operator, H::AbstractOperator, J, method! = nothing, args...;
+                    rates=nothing, Jdagger=dagger.(J), kwargs...)
+
+    if method! === nothing
+        if isblascompatible(rho0) && isblascompatible(H) && all(isblascompatible.(J))
+            _method! = IterativeSolvers.bicgstabl!
+        else
+            _method! = IterativeSolvers.idrs!
+        end
+    else
```

Comment:
On a second note, I noticed that since version 0.9 of `IterativeSolvers`, they removed their explicit calls to BLAS and started using 5-arg `mul!`. The relevant commit is [this one](https://github.com/JuliaLinearAlgebra/IterativeSolvers.jl/commit/39363bb5e1efd77bcbd9f3f3038f214b666d7782)

So this switch is no longer needed (`bicgstabl!` should work also for non-blas types) and you could just use `IterativeSolvers.bicgstabl!` as a default. @Z-Denis , I assume that is the best solver to have as a default?

Reply:
Thanks for pointing that out! It appears to work fine with `bicgstabl!` for all cases. Let's wait for @Z-Denis to comment on the default choice here.

Reply:
> So this switch is no longer needed (`bicgstabl!` should work also for non-blas types) and you could just use `IterativeSolvers.bicgstabl!` as a default. @Z-Denis , I assume that is the best solver to have as a default?

That's nice. Indeed, `SteadyState.jl` was falling back to idrs! because of this explicit BLAS call. Because of this, I never really extensively benchmarked both solvers on the same kind of problem. The choice was made on the basis of a discussion with the authors of [this paper](https://journals.aps.org/prl/abstract/10.1103/PhysRevLett.122.110405), who had used it extensively therein.